From 349cffa03f4899026f6f0e37da569b9aa0f391ca Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Sun, 5 Feb 2017 15:17:57 +0200
Subject: [PATCH 36/58] ARM: i.MX6: dts: Add initial support for SOM-iMX6

Add initial support for SOM-iMX6 based on a FreeScale imx6q SoC.

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 arch/arm/boot/dts/imx6q-cl-som-imx6.dts  | 22 ++++++++++++++++++++++
 arch/arm/boot/dts/imx6q-cl-som-imx6.dtsi | 14 ++++++++++++++
 arch/arm/boot/dts/imx6q-sbc-imx6.dts     | 24 ++++++++++++++++++++++++
 3 files changed, 60 insertions(+)
 create mode 100644 arch/arm/boot/dts/imx6q-cl-som-imx6.dts
 create mode 100644 arch/arm/boot/dts/imx6q-cl-som-imx6.dtsi
 create mode 100644 arch/arm/boot/dts/imx6q-sbc-imx6.dts

diff --git a/arch/arm/boot/dts/imx6q-cl-som-imx6.dts b/arch/arm/boot/dts/imx6q-cl-som-imx6.dts
new file mode 100644
index 0000000..582bdd5
--- /dev/null
+++ b/arch/arm/boot/dts/imx6q-cl-som-imx6.dts
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2016 CompuLab Ltd.
+ *
+ * Author: Valentin Raevsky <valentin@compulab.co.il>
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+/dts-v1/;
+#include "imx6q.dtsi"
+#include "imx6q-cl-som-imx6.dtsi"
+#include "imx6q-cm-fx6.dtsi"
+
+/ {
+	model = "CompuLab SOM-iMX6";
+	compatible = "compulab,som-imx6", "compulab,cm-fx6", "fsl,imx6q";
+};
diff --git a/arch/arm/boot/dts/imx6q-cl-som-imx6.dtsi b/arch/arm/boot/dts/imx6q-cl-som-imx6.dtsi
new file mode 100644
index 0000000..ee87f1b
--- /dev/null
+++ b/arch/arm/boot/dts/imx6q-cl-som-imx6.dtsi
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2016 CompuLab Ltd.
+ *
+ * Author: Valentin Raevsky <valentin@compulab.co.il>
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#define __SOM_iMX6__
diff --git a/arch/arm/boot/dts/imx6q-sbc-imx6.dts b/arch/arm/boot/dts/imx6q-sbc-imx6.dts
new file mode 100644
index 0000000..42af61d
--- /dev/null
+++ b/arch/arm/boot/dts/imx6q-sbc-imx6.dts
@@ -0,0 +1,24 @@
+/*
+* Copyright 2016 CompuLab Ltd.
+*
+* Author: Valentin Raevsky <valentin@compulab.co.il>
+*
+* The code contained herein is licensed under the GNU General Public
+* License. You may obtain a copy of the GNU General Public License
+* Version 2 or later at the following locations:
+*
+* http://www.opensource.org/licenses/gpl-license.html
+* http://www.gnu.org/copyleft/gpl.html
+*/
+
+/dts-v1/;
+#include "imx6q.dtsi"
+#include "imx6q-cl-som-imx6.dtsi"
+#include "imx6q-cm-fx6.dtsi"
+#include "imx6qdl-sb-fx6x.dtsi"
+#include "imx6qdl-sb-fx6.dtsi"
+
+/ {
+	model = "CompuLab SOM-iMX6 on SBC-FX6";
+	compatible = "compulab,som-imx6", "compulab,cm-fx6", "compulab,sbc-fx6", "fsl,imx6q";
+};
-- 
1.9.1

